lib/charms/loki_k8s/v0/log_proxy.py

```diff
@@ -321,12 +338,86 @@ def _create_directories(self) -> None:
         self._container.make_dir(path=WORKLOAD_CONFIG_DIR, make_parents=True)
 
     def _obtain_promtail(self, event) -> None:
-        if self._is_promtail_binary_in_workload():
+        """Obtain promtail binary from an attached resource or download it."""
+        if self._is_promtail_attached():
+            return
```

Comment:
Is there an easy way to get a general "pushable" (`open(BINARY_PATH, "rb")` or `open(resource_path, "rb")`) and do `container.push` only once?

Reply:
I have a workaround:

```python
    def _push_binary_to_workload(self, resource_path=BINARY_PATH) -> None:
        with open(resource_path, "rb") as f:
            self._container.push(WORKLOAD_BINARY_PATH, f, permissions=0o755, make_dirs=True)
            logger.debug("Promtail binary file has been pushed to workload container.")
```